include/nbl/builtin/hlsl/limits.hlsl

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_LIMITS_INCLUDED_
+#define _NBL_BUILTIN_HLSL_LIMITS_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+
+// C++ headers
+#ifndef __HLSL_VERSION
+#include <limits>
+#endif
+
+namespace nbl
+{
+namespace hlsl
+{
+
+#ifdef __HLSL_VERSION
+
+enum float_denorm_style {
+    denorm_indeterminate = -1,
+    denorm_absent        = 0,
+    denorm_present       = 1
+};
+
+enum float_round_style {
+    round_indeterminate       = -1,
+    round_toward_zero         = 0,
+    round_to_nearest          = 1,
+    round_toward_infinity     = 2,
+    round_toward_neg_infinity = 3
+};
```

Comment:
if these exist in `std` then we need an alias in the C++ side, or we need to define them outside